src/app/carbon-estimation/carbon-estimation.component.ts

```diff
@@ -48,12 +43,20 @@ export class CarbonEstimationComponent implements OnInit, OnDestroy {
 
   constructor(private changeDetectorRef: ChangeDetectorRef) {
     effect(() => {
-      this.emissions = this.getOverallEmissionPercentages(this.carbonEstimation());
-      this.emissionAriaLabel = this.getAriaLabel(this.emissions);
+      if (this.showPlaceholderDiagram) {
```

Comment:
I wonder if it would be better to just use the carbonEstimation input here - store it locally before the if. That way Typescript should know that it can't be null when you set it in the chartData with the ! added.

Reply:
Yes, that seems neater. I've made that change.